common.go

```diff
@@ -32,32 +32,16 @@ func listenForInterrupt(q chan struct{}) {
 }
 
 func kafkaVersion(s string) sarama.KafkaVersion {
-	dflt := sarama.V0_10_0_0
-	switch s {
-	case "v0.8.2.0":
-		return sarama.V0_8_2_0
-	case "v0.8.2.1":
-		return sarama.V0_8_2_1
-	case "v0.8.2.2":
-		return sarama.V0_8_2_2
-	case "v0.9.0.0":
-		return sarama.V0_9_0_0
-	case "v0.9.0.1":
-		return sarama.V0_9_0_1
-	case "v0.10.0.0":
+	if s == "" {
 		return sarama.V0_10_0_0
-	case "v0.10.0.1":
-		return sarama.V0_10_0_1
-	case "v0.10.1.0":
-		return sarama.V0_10_1_0
-	case "v0.10.2.0":
-		return sarama.V0_10_2_0
-	case "":
-		return dflt
 	}
 
-	failf("unsupported kafka version %#v - supported: v0.8.2.0, v0.8.2.1, v0.8.2.2, v0.9.0.0, v0.9.0.1, v0.10.0.0, v0.10.0.1, v0.10.1.0, v0.10.2.0", s)
-	return dflt
+	v, err := sarama.ParseKafkaVersion(strings.TrimPrefix(s, "v"))
+	if err != nil {
+		failf(err.Error())
```

Comment:
just a quick note: can you make sure these are legible to a cli user? ie, does the error message give you enough context that the user would know that the version string they're passing isn't correct/supported? also, changing this means that we generally want to support all versions that sarama supports - i think that's not true yet, or? cf #83

Reply:
if version regex velidation fail, ParseKafkaVersion return `invalid version %s` as error, we can add more context if you want.

Since we use sarama for operations on kafka, we should be able to support all versions, sarama supports. Kafka 2.0 is not working in master since sarama 1.12 does not support it.

Reply:
> if version regex velidation fail, ParseKafkaVersion return invalid version %s
alright, sounds good to me. 

> Kafka 2.0 is not working in master since sarama 1.12 does not support it.
alright, i would like to make sure that happens anyway, so let's pull this update in and then update the tests on master to run against newer versions of kafka.